snuba/web/rpc/v1/endpoint_get_traces.py

```diff
@@ -0,0 +1,289 @@
+        original_body=MessageToDict(request),
+        query=_build_query(request),
+        query_settings=query_settings,
+        attribution_info=AttributionInfo(
+            referrer=request.meta.referrer,
+            team="eap",
+            feature="eap",
+            tenant_ids={
+                "organization_id": request.meta.organization_id,
+                "referrer": request.meta.referrer,
+            },
+            app_id=AppID("eap"),
+            parent_api="eap_span_samples",
+        ),
+    )
+
+
+def _convert_results(
+    request: GetTracesRequest, data: Iterable[Dict[str, Any]]
+) -> list[GetTracesResponse.Trace]:
+    converters: Dict[str, Callable[[Any], AttributeValue]] = {}
+
+    for trace_column in request.columns:
+        if trace_column.type == AttributeKey.TYPE_BOOLEAN:
+            converters[trace_column.name] = lambda x: AttributeValue(val_bool=bool(x))
+        elif trace_column.type == AttributeKey.TYPE_STRING:
+            converters[trace_column.name] = lambda x: AttributeValue(val_str=str(x))
+        elif trace_column.type == AttributeKey.TYPE_INT:
+            converters[trace_column.name] = lambda x: AttributeValue(val_int=int(x))
+        elif trace_column.type == AttributeKey.TYPE_FLOAT:
+            converters[trace_column.name] = lambda x: AttributeValue(val_float=float(x))
+
+    res: list[GetTracesResponse.Trace] = []
+    column_ordering = {
+        trace_column.name: i for i, trace_column in enumerate(request.columns)
+    }
+
+    for row in data:
+        values: defaultdict[
+            TraceColumn.Name,
+            GetTracesResponse.Trace.Column,
+        ] = defaultdict(GetTracesResponse.Trace.Column)
+        for column_name, value in row.items():
+            name = _NAME_TO_COLUMN[column_name]
+            if name in converters.keys():
+                values[name] = GetTracesResponse.Trace.Column(
+                    name=name,
+                    value=converters[name](value),
+                )
+        res.append(
+            GetTracesResponse.Trace(
+                # we return the columns in the order they were requested
+                columns=sorted(
+                    values.values(),
+                    key=lambda c: column_ordering[c.name],
+                )
+            )
+        )
+
+    return res
+
+
+def _get_page_token(
+    request: GetTracesRequest, response: list[GetTracesResponse.Trace]
+) -> PageToken:
+    if not response:
+        return PageToken(offset=0)
+    num_rows = len(response)
+    return PageToken(offset=request.page_token.offset + num_rows)
+
+
+def _validate_order_by(in_msg: GetTracesRequest) -> None:
+    order_by_cols = set([ob.column.name for ob in in_msg.order_by])
+    selected_columns = set([c.name for c in in_msg.columns])
+    if not order_by_cols.issubset(selected_columns):
+        raise BadSnubaRPCRequestException(
+            f"Ordered by columns {order_by_cols} not selected: {selected_columns}"
+        )
```

Comment:
Is the intention here to allow ordering by any of the selected columns? The 2 main ones we'd like to have will be by timestamp (newest/oldest trace) and by duration (shortest/longest trace) though this second one is harder to define clearly as the trace duration is as it's becoming closer to a session depending on the platform.

I'm not opposed to just supporting ordering by timestamp first and we can add more later once we more clearly define them.